src/LiveQuerySubscription.js

```diff
@@ -101,16 +101,19 @@ class Subscription extends EventEmitter {
     this.id = id;
     this.query = query;
     this.sessionToken = sessionToken;
+
+    // adding listener so process does not crash
+    // best practice is for developer to register their own listener
+    this.on('error', () => {});
   }
 
   /**
-   * closes the subscription
+   * Closes the subscription
    */
-  unsubscribe() {
-    return CoreManager.getLiveQueryController().getDefaultLiveQueryClient().then((liveQueryClient) => {
-      liveQueryClient.unsubscribe(this);
-      this.emit('close');
-    });
+  async unsubscribe(): Promise {
+    const liveQueryClient = await CoreManager.getLiveQueryController().getDefaultLiveQueryClient();
+    liveQueryClient.unsubscribe(this);
```

Comment:
Should we wait for this to unsubscribe or just let it fire off in the background?

```
this.connectPromise.then(() => {
  this.socket.send(JSON.stringify(unsubscribeRequest));
});
```
I don't see any issues with the client itself but wanted feedback because `this.connectPromise.then()` is used a lot in the client.

Reply:
In my old prototype, I waited for all subs to unsubscribe, then called Parse.LiveQuery.close(), I guess it was useless after all, since Parse.LiveQuery.close() is suposed to do it by itself.

Reply:
I'd think you'd wait for the promise to resolve before firing the event, unless there's a goog reason not to?

Reply:
I'm going to revert this function. Since a socket is being used we have no way to tell if it went through or not (it will eventually). 

As I have it, the caller is now forced to wait for the unsubscribe to finish (which wasn't the case before).